src/main/java/org/junitpioneer/jupiter/EnvironmentVariableUtils.java

```diff
@@ -95,7 +95,19 @@ private static void setInSystemEnvClass(Consumer<Map<String, String>> consumer)
 	private static Map<String, String> getFieldValue(Class<?> clazz, Object object, String name)
 			throws ReflectiveOperationException {
 		Field field = clazz.getDeclaredField(name);
-		field.setAccessible(true); //NOSONAR illegal access required to implement the extension
+		try {
+			field.setAccessible(true); //NOSONAR illegal access required to implement the extension
+		}
+		catch (RuntimeException ex) {
+			// Java 9 added InaccessibleObjectException (but this project compiles against Java 8 at the moment)
+			if (ex.getClass().getName().equals("java.lang.reflect.InaccessibleObjectException")) {
+				throw new ExtensionConfigurationException("Cannot access JDK internals to modify environment variables."
+						+ " Have a look at the Javadoc or the documentation for possible solutions.",
+					ex);
+			} else {
+				throw ex;
```

Comment:
We could send the user directly to the corresponding part in the documentation:

https://junit-pioneer.org/docs/environment-variables/#warnings-for-reflective-access

Also, Nicolai probably prefers an `if` without curly braces. 😉

```suggestion
			// Java 9 added InaccessibleObjectException, but we compile against Java 8.
			if (ex.getClass().getName().equals("java.lang.reflect.InaccessibleObjectException"))
				throw new ExtensionConfigurationException("Cannot access Java runtime internals to modify environment variables. "
						+ "Have a look at the documentation for possible solutions: "
						+ "https://junit-pioneer.org/docs/environment-variables/#warnings-for-reflective-access",
					ex);
			throw ex;
```